### PR TITLE
Add a test verifying that etcd watches work even when there's a bunch of unrelated changes

### PR DESCRIPTION
**Pull request by lavalamp**
This passes for me locally. Suggests that #6059's problem is our use of etcd, not etcd itself.

### CODE REVIEW

**Comment by dchen1107** [comment]
Based on your description, the test itself couldn't reproduce the watch delay at all? Since you suggested that issue should be in the way on our use of etcd, can we make the test simulating how we are using etcd?

**Comment by lavalamp** [comment]
The point of this test is to rule out etcd as a source of the problem.

...hm, maybe I could make a more advanced test that checks our entire system. That would be interesting.

**Comment by dchen1107** [comment]
Thanks for the effort on reproducing the issue. Now we might have to disable the test by default, so that it can pass, be merged, and everyone can play with it. :-P

**Comment by lavalamp** [comment]
Test is disabled by default now.

**Comment by lavalamp** [comment]
...and I removed the test that didn't repro the issue. should be good to go now.

File: test/integration/client_test.go [hunk 1]
```diff
@@ -132,3 +145,214 @@ func TestClient(t *testing.T) {
 		}
 	}
 }
+
+func TestMultiWatch(t *testing.T) {
+	// Disable this test as long as it demonstrates a problem.
```
Comment (dchen1107) [inline test/integration/client_test.go]: nit: I would rather have a TODO: enable the test once the problem is resolved. Up to you on this.

**Comment by dchen1107** [comment]
LGTM except one nit. Will merge once travis / shippable is green.

**Pull request merged**